buildSrc/src/main/kotlin/toolkit-integration-testing.gradle.kts

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+plugins {
+    id("java")
+    id("idea")
+    id("toolkit-testing")
+}
+
+val integrationTests: SourceSet = sourceSets.maybeCreate("integrationTest")
+sourceSets {
+    integrationTests.apply {
+        java.setSrcDirs(listOf("it"))
+        resources.srcDirs(listOf("it-resources"))
+
+        compileClasspath += main.get().output + test.get().output
+        runtimeClasspath += main.get().output + test.get().output
+    }
+}
+
+configurations.getByName("integrationTestImplementation") {
+    extendsFrom(configurations.getByName(JavaPlugin.TEST_IMPLEMENTATION_CONFIGURATION_NAME))
+    isCanBeResolved = true
+}
+configurations.getByName("integrationTestRuntimeOnly") {
+    extendsFrom(configurations.getByName(JavaPlugin.TEST_RUNTIME_ONLY_CONFIGURATION_NAME))
+    isCanBeResolved = true
+}
+
+// Add the integration test source set to test jar
+val testJar = tasks.named<Jar>("testJar") {
+    from(integrationTests.output)
+}
+
+// Silly but allows higher throughput of the build because we can start compiling / testing other modules while the tests run
+tasks.test {
```

Comment:
Why is this needed in integration testing as well? Add a comment?

Reply:
Ya, I will clarify it